### ISSUE TITLE
Logs are all written to `console.error` and overriding the logger only partially works

### ISSUE THREAD

**Issue by dennisameling** [grpc/grpc-node#2594]
### Problem description

In this library's `DEFAULT_LOGGER`, all logs are written to `console.error`, no matter whether it's an `error`, `info`, or `debug` log:

https://github.com/grpc/grpc-node/blob/83743646cf69baf9ae1294015de5ffed33339154/packages/grpc-js/src/logging.ts#L20-L30

I would expect the library to write to the appropriate channel instead (`error`, `info`, or `debug`).

### Reproduction steps

Just set `GRPC_TRACE` to `all` and `GRPC_VERBOSITY` to `DEBUG`. Tools like Datadog will indicate that **all** messages coming from `grpc-js` are errors, while they aren't.

### Environment
 - OS name, version and architecture: macOS 14.0 x86_64
 - Node version [e.g. 8.10.0] 18.16.0
 - Node installation method [e.g. nvm] nvm
 - If applicable, compiler version [e.g. clang 3.8.0-2ubuntu4] N/A
 - Package name and version [e.g. gRPC@1.12.0] grpc-js@1.9.5

### Additional context

I've overridden the logger as follows:

```TS
import {setLogger} from '@grpc/grpc-js'

setLogger({
    error: (message?: any, ...optionalParams: any[]) => {
        console.error('E ' + message, ...optionalParams);
    },
    info: (message?: any, ...optionalParams: any[]) => {
        console.info('I ' + message, ...optionalParams);
    },
    debug: (message?: any, ...optionalParams: any[]) => {
        console.debug('D ' + message, ...optionalParams);
    },
})
```

However, even when overriding the logger, the `Loading @grpc/grpc-js version XXX` message is still written to `console.error`:

```
2023-10-10T09:10:03.469Z	undefined	ERROR	D 2023-10-10T09:10:03.469Z | index | Loading @grpc/grpc-js version 1.9.5
```

https://github.com/grpc/grpc-node/blob/83743646cf69baf9ae1294015de5ffed33339154/packages/grpc-js/src/index.ts#L281-L285

This is because the TypeScript compilation process turns our code into `commonjs`, meaning that the log line above is triggered before we are even able to override the logger:

```JS
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });

// This call triggers the "Loading @grpc/grpc-js version XXX" message before the logger is overridden
const grpc_js_1 = require("@grpc/grpc-js");

(0, grpc_js_1.setLogger)({
    error: (message, ...optionalParams) => {
        console.error('E ' + message, ...optionalParams);
    },
    info: (message, ...optionalParams) => {
        console.info('I ' + message, ...optionalParams);
    },
    debug: (message, ...optionalParams) => {
        console.debug('D ' + message, ...optionalParams);
    },
});
```

Could you please consider:
- Setting the correct console log type for the corresponding log levels (`console.error`, `console.info`, `console.debug`)
- Updating the code in such way that the `Loading @grpc/grpc-js version XXX` message isn't triggered before we get a chance to override the logger

Thanks! Let me know if I can help with anything.

**Comment by murgatroid99** [comment]
The original intention of that logger was to emulate the behavior of the gRPC C core library that the old `grpc` library was built on, and in that library all log messages were written to stderr. I would be open to changing it, but since you can customize it (other than for this one log message), I don't think it's too necessary.

Considering the log message itself, it is useful to see what version a log came from, but it doesn't have to be in that spot. One useful alternative I can see is including the version in the common trace log prefix and removing that log line entirely.

For now, to mitigate this, I recommend using the trace setting `GRPC_TRACE=all,-index` to disable that one log line.

**Comment by dennisameling** [comment]
Thanks for your quick reply! That makes sense. Can confirm `GRPC_TRACE=all,-index` works as expected 👍🏼 

If you think adding it to the common trace log prefix is the best way forward, I'd be happy to provide a PR for that.

**Comment by murgatroid99** [comment]
I published grpc-js version 1.9.6 with that `index` tracer removed.
